---
  title: "Thesis_documentation"
output: html_document
---

Thesis proposal/Outline

### The problem background:
	Biodiversity in the Arctic is unique, and its microbe community structure is distinct from other marine environment. The uniqueness of the biodiversity in the Arctic comes from several environmental variables such as the low air temperature, the high salinity of the surface waters, and the restricted exposure of sunlight from the limited annual light cycle and sea ice. Particularly, the light cycle in the Arctic plays a significant role in determining the microbial community structure throughout the year. For instance, when the light intensity increases and excessive nutrients such as nitrogen and phosphorus are introduced from the melting sea ice typically during the early spring, the phytoplankton population rapidly increases and reaches to the peak in abundance, exporting vertical particulate organic carbons composed of the settlements of phytoplankton cells and zooplankton faecal pellets. The phytoplankton diversity of the blooming period can be important to measure the carbon cycle and seasonal shift of biodiversity of the marine environment. To enhance our knowledge on the composition and temporal dynamic of the eukaryotic diversity during these carbon exporting periods in the Arctic, global comparison of eukaryotic microbial diversity of POC is facilitated and a baseline for community changes during the carbon exporting events is determined. 

### Research Question:
	In my bachelor thesis project, I will be working on the global comparison of eukaryotic diversity to characterize a baseline for the community shift during the peak of vertical carbon fluxes in the Framstrait in the Arctic Ocean. This project is collanborated with the work of master theisis project by Raissa Meyer and doctorial thesis project by Magda in which the analysis of eukaryotic diversity in the Fram Strait complements with the prokaryotic diversity analysis by Raissa. 
	
	
### Methods:
The eukaryotic microbial diversity of the Arctic will be analyzed using 18s rRNA gene sequence data from PeakPOC datasets. The sequence data will be processed with DADA2 to infer the exact amplicon sequence variants and produce a finer scale resolution, avoiding the variablility of output results caused by different OTU clustering algorithms. The output ASV table will further be used to measure alpha and beta diversity and analyzed with multivariate statistic approach. I will also apply centered log ratio transformation (CLR) for number count of the ASVs.  In the end, the global comparison will be carried out with published studies that investigated eukaryotic microbial from a time series of algal blooms in the multiple ocean observatories. The comparison of results includes the differences of stability of alpha diversity in each dataset, detection of abundant species from each sample, and the identification of key species responsible for the algal bloom.  

### Publication 1
**doi** [link](https://doi. org/10.1371/journal.pone.0196987)
**Title** "Diversity and potential activity patterns of planktonic eukaryotic microbes in a mesoeutrophic coastal area (Eastern English Channel)"
**Summary** The diversity of planktonic eukaryotic microbes was studied at a coastal station of the eastern English Channel (EEC) from March 2011 to July 2015 which is a span of 5 years (77 samples), using highthroughput sequencing of the **V2-V3** hypervariable region of the 18S SSU rDNA gene. The results showed that eight super-groups were identified throughout all samples: _Alveolata_, _Stramenopiles_, _Opisthokonta_, _Hacrobia_, _Archeaplastida_, _Apusozoa_, _Rhizaria_, and _Amoebozoa_. At the end of winter, the abundance of nitrate and high N:P and N:Si ratios triggered blooms of _P.globosa_, following by rises of colonial diatoms and heterotrophic dinoflagellates. Microbial eukaryote diversity was at the lowest during P. globosa bloom period, but gradually increased to reach its highest in fall-winter. Symbionts and degraders accounted for more than one third of the OTU diversity in fall-winter

### Publication 2 
**doi** [link] (https://doi.org/10.3389/fmicb.2019.02722)
**Title** “Eukaryotic Phytoplankton Contributing to a Seasonal Bloom and Carbon Export Revealed by Tracking Sequence Variants in the Western North Pacific”
**Summary** This study examined the eukaryotic phytoplankton communities from the surface to abyssopelagic zone throughout the year in the western subtropical North Pacific. In this sampling point, two periods of oceanographic conditions were observed. One is stratification which limits photosynthetic processes by creating barriers to nutrient mixing, and the other one is mixing period. The results showed that tax related to prasinophytes were observed throughout the year and likely played a role in the seasonal bloom that occurred during the mixing period. Additionally, abundance of mamiellophyceae in the abyssopelagic zone suggested that prasinophytes are involved in carbon export. 

- A year long experiment.
- Samples were taken every 4 months from Nov 2010 to July 2011. (nov, feb apr, Jul)
- Samples were collected from different depths of water from 0(surface) to 5000m (abyssopelagic zone)
- Stratification period = Nov and July
- Mixing period = Feb and April







### 11/03/2020

I finished running a DADA2 pipeline for PeakPOC datasets. At this moment, I am currently working on assigning eukaryotic taxonomy to the 18s ASVs infered from DADA2. I will have to manually upload it to the SILVAngs and create a taxonomy table. 

On top of this, I need to add environmental data to the metadata for PeakPOC datasets. This can be easily done by following magda's script. 

As of now, I am stuck in retriving raw sequence files from EEC paper. The sequence file is multiplexed with 33 barcode sequences which some of the barcode sequences are identical, and it makes difficult to distinguish separate files. To solve this, I found this tool called deML which distinguishes two samples that have identical barcode sequences https://github.com/grenaud/deMLhttps://github.com/grenaud/deML


### 12/03/2020

Today's task was to produce a metadata table for PeakPOC samples using a script given from Magda. Although the scrip was written in a logical flow, it failed to add the desired value from input numbers. On top of this, I attempted to enhance the efficiency of this workload by creating a forloop script to automate the process. However, it also failed to add values in each row in the output file. At the end, I manually inserted the value in each row by hand, which would have been faster if I have known that I would spend almost 3 hours on figuring out the problem of this script. 

### 13/03/2020

Today's task was to upload an ASVs table of PeakPOC dataset onto silvangs platform to asssign taxonomy to each unique sequenceses. The problem I encounter was that I kept receiving error messages stating that the spaces between names of unique sequence IDs are causing to fail the process. I forwarded this problem to Raissa and she helped me with her command line that she previously used to solve the same issue. 

```{engine='bash'}
cat ASVs.fa | tr -d " " > ASVs_nospaces.fa
```

This removes the spaces in the sequence IDs.


### 14/03/2020

I received an email from the author of EEC paper. He mentioned that the sequencing company had two separate runs for their samples. This was the reason for having two identical barcode tag in two different samples. THe problem was occured because NCBI merged two separate runs together, causing it impposible to separate by indivudal samples. He also suggested that he can upload flow files on a ftp server for me to reanalyze it. I also had problem setting up this ftp server. At the end, I just sent him my one drive link to upload it to my folder directly. 

### 15/ 03/2020

I smoothly downloaded all the sequence files for the dataset from Japanese coast study. Right now I am running these files to DADA2. THe only concern I have is that these files sum up to a total of 8GB which is relatively large compared to other datasets I am using in this study. 

### 17/03/2020

I struggled demultiplexing the fastq file from EEC datasets for 2011-2013. Since this fastq file was sequenced in 454 pyrosequencing platform and the two of the runs were merged together on the NCBI database, it was impossible to demultiplex by the barcode tag listed on the description page. Fortunately, I received a response from the author of this paper, and the author provided me with the flow files for individual samples. By using unique identifer IDs from each flow file, I was able to demultiplex the fastq file using a tool called "fastq-grep". The pipeline for demultiplexing fastq file is uploaded on the github repository. 

### 18/03/2020

Another problem that I encounter from DADA2 was that the forward reads and the reverse reads from the Japanese datasets did not successfully merge together. As I looked thorugh the answer from Github issue on DADA2, I found that the number of overlapping reads between the forward file and the reverse was insufficient. I proceeded with just by using forward reads and I will come back to this issue once I catch up with my other work. 


### 20/03/2020

Sample selection for each publication

PeakPOC ---- 25 samples collected in spring and summer.

EEC ---- I filtered 15 samples from 2011-2013 sequenced by 454 pyrosequencing and 15 samples from 2013-2015 sequenced by illumina Miseq. The selection was made based on the relevance to the result discussed in the original paper, including blooms being observed during the early spring and late summer, and late winter. Metadata regarding the samples selected from the EEC dataset are in a file ""

Japan ----- I selected sequence reads from 0m to 200m since chlorophyl concentration was measured within this depth range, and also the discussion of this paper focused on the community diversity of thw shallower water columns. Metadata regarding this Japanese dataset are contained in a file "SraRunTable_japan_without100m". 100m was excluded due to the large file size and ease computational load for my computer. 


### 22/03/2020

I moved forward to analysis part of this project. All the generated files from the PeakPOC dataset were assigned into a phyloseq object, and the alpha diversity plots and rarefaction curves were generated. I received few feedback from Magda regarding the plots, which are

- Change the name of file to more meaningful and understandable one
- Rearrange x axis by dates instead of season
- Data transformation method
- Try to create an NMDS plot. 


### 25/03/2020

- ANOVA like differential expression (ALDEx) Analysis for Mixed population RNA-Seq

Problem; The single error term accounts for variation due to three sources: sampling, tehnical replication, and persample variability. 
the main purpose of this work is to show that within-cibdutuib variation cannot reasonably be ignored, especially in Meta-RNA-Seq experinebts, and if such variation is not correcly accomodatedfor then misleading inference will occur. 
THe data are modeled as compositional or proportional (not relative abundance) because the reads obtained on a high-throughput sequencing run are constrained to the total number of reads available. 

The standard 16S tag-sequencing workflow normalizes abundances between samples by rarefaction or other subsampling methods and usually works with proportions. 
Datasets can be proportiona;, such as fractions of the whole, percentages, parts per million.
There have been many warnings regarding the use of standard statistical methods that assume the independence of the underluing observations when examining compositional data
The m

- Instead of comparing the composition of count values within the sample, the ratio between proportions is compared to have robust data interpretaations.
- Centered log ratio transformation - the read counts for each ASV are divided by the mean of count of ASVs within the sample, followed by taking the logarithm. 
CLR has the advantage that there is a one to one transformation of all ASVs. 
- 2 is used as the base of the logarithm, implying that the differences between ASVs represent fold changes in relative abundance between ASVs 



### 25/03/2020

LINE 113 
https://rstudio-pubs-static.s3.amazonaws.com/7433_4537ea5073dc4162950abb715f513469.html
This page explains how to reorder x-axis from other variables. In this case, I reorder x-axis in a way that the sample names are reordered based on chronological order from the vairable "Collection_Date"

### 27/03/2020

- NMDS was pltoted based on this tutorial https://jkzorz.github.io/2019/06/06/NMDS.html https://rpubs.com/brownjk5/NMDS https://jonlefcheck.net/2012/10/24/nmds-tutorial-in-r/ 

- Reordering items from legend labels has been done through this tutorial https://stackoverflow.com/questions/38619402/how-to-reorder-the-items-in-a-legend

- Changing the shapes for different variables https://www.datanovia.com/en/blog/ggplot-point-shapes-best-tips/

### 28/03/2020

- Merging two tables by rownames. https://statisticsglobe.com/r-merge-data-frames-by-row-names 

### 30/03/2020

I fixed label names and colors for hieracheal clustering dendrogram plot. Color manipulation was inspired by this tutorialhttp://www.sthda.com/english/wiki/beautiful-dendrogram-visualizations-in-r-5-must-known-methods-unsupervised-machine-learning

